portal-kernel/src/com/liferay/portal/kernel/util/MapUtil.java

```diff
@@ -346,7 +346,7 @@ public static String getString(
 
 		V value = map.get(key);
 
-		if (value != null) {
+		if ((value instanceof String) && Validator.isNotNull((String)value)) {
```

Comment:
Right now, the function is only used with `Map<Locale, String>` but the function uses generics, so maybe is a good idea check the value != null too without being String or not to prevent future bugs.

Reply:
Hi @AliciaGarciaGarcia,

Do you mean: 
`if ((!(value instanceof String) && value != null) || ((value instanceof String) && Validator.isNotNull((String)value))) {
			return value;
		}`

It looks ugly, but we need to be sure that we are not checking String with null, otherwise we will get back this issue.

Regards.

Reply:
you can always invert the if to be more _attractive_?

```
	if (value == null || (value instanceof String && Validator.isBlank((String)value) )) {
		return map.get(fallbackKey);
	}

	return value;
```

Reply:
yes, I agree, change done!